game_template_1.0

```diff
@@ -0,0 +1,264 @@
+class MedicalBay : public Room
+{
+public:
+    // Function to handle entering the medical bay
+    void enter(Copilot& copilot, bool& enginesOn, bool& hasTool) override
+    {
+        cout << readFromFile("medical_bay.txt");
+        if (copilot.obtained && copilot.injured)
+        {
+            cout << readFromFile("copilot_healed.txt");
+            copilot.injured = false; // Heal the copilot
+        }
+        else
+        {
+            cout << readFromFile("nothing_to_do.txt");
+        }
+    }
+};
+
+// Class for the Storage Room
+class StorageRoom : public Room
+{
+public:
+    // Function to handle entering the storage room
+    void enter(Copilot& copilot, bool& enginesOn, bool& hasTool) override
+    {
+        if (!copilot.obtained)
+        {
+            cout << readFromFile("storage_locked.txt");
+            return; // Prevent entry if copilot is not found
+        }
+
+        cout << readFromFile("storage_room.txt");
+        cout << readFromFile("enemy_appears.txt");
+
+        int totalRoll = 0;
+        for (int i = 0; i < 3; ++i)
+        {
+            waitForKeyPress();
+            int roll = rand() % 20 + 1;
+            cout << "You rolled a " << roll << ".\n";
+            totalRoll += roll;
+        }
+
+        if (totalRoll >= 40)
+        {
+            cout << readFromFile("enemy_defeated.txt");
+        }
+        else
+        {
+            cout << readFromFile("copilot_injured.txt");
+            copilot.injured = true; // Mark copilot as injured
+        }
+
+        cout << readFromFile("tool_found.txt");
+        hasTool = true; // Mark tool as found
+    }
+};
+
+// Class for the Engine Room
+class EngineRoom : public Room
+{
+public:
+    // Function to handle entering the engine room
+    void enter(Copilot& copilot, bool& enginesOn, bool& hasTool) override
+    {
+        cout << readFromFile("engine_room.txt");
+        if (!hasTool)
+        {
+            cout << readFromFile("need_tool.txt");
+            return; // Prevent puzzle solving if tool is not found
+        }
+
+        cout << readFromFile("solving_puzzle.txt");
+        // Simple puzzle simulation
+        cout << readFromFile("puzzle_solved.txt");
+        enginesOn = true; // Mark engines as fixed
+    }
+};
+
+int main()
+{
+    // Initialize game state
+    Copilot copilot; // Create an instance of Copilot
+    bool enginesOn = false; // Track if engines are fixed
+    bool hasTool = false; // Track if the player has the tool
+
+    // Initialize room objects
+    Cockpit cockpit;
+    SleepingQuarters sleepingQuarters;
+    MedicalBay medicalBay;
+    StorageRoom storageRoom;
+    EngineRoom engineRoom;
+
+    srand(static_cast<unsigned int>(time(0))); // Seed the random number generator
+
+    // Display welcome message and introduction
+    cout << "Welcome to the Untitled RPG game!\n";
+    cout << readFromFile("introduction.txt");
+
+    // Main game loop
+    while (true)
+    {
+        // Display room options
```

Comment:
Hoping to add this functionality to my assigned issue as a Class. That may allow for some flexibility if we determine we want to add other features. The class would be Hallway, inheriting from the Room class.